demos/es7/advertised-echo.js

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env node
+const Bot = require('../../lib/index.js')
+
+const bot = new Bot()
+
+async function main() {
+  try {
+    const username = process.env.KB_USERNAME
+    const paperkey = process.env.KB_PAPERKEY
+    await bot.init(username, paperkey)
+    const info = bot.myInfo()
+    console.log(`Echo bot initialized with username ${info.username}.`)
+
+    await bot.chat.clearCommands()
+    await bot.chat.advertiseCommands({
+      advertisements: [
+        {
+          type: 'public',
+          commands: [
+            {
+              name: '!echo',
+              description: 'Sends out your message to the current channel.',
+              usage: '[your text]',
+            },
+          ]
+        }
+      ],
+    })
+
+    const onMessage = async message => {
+      if (message.content.type !== 'text') {
+        return
+      }
+
+      if (!message.content.text.body.startsWith('!echo ')) {
```

Comment:
Would it make sense to modify the watch methods so that they automatically ignore non-command messages and ignore the prefixes?

Reply:
hmm, what if the bot registers commands but also has access to all messages? I'd say either it should work like it does here, or, we add a pretty wrapper around doing this extra work and doesn't complicate the inside of any of our existing `watch` functions.

so here you could have:

```
await bot.chat.clearCommands()
await bot.chat.advertiseCommands(/*stuff*/)
bot.chat.watchForCommand('!echo', (message, argumentText) => {
 bot.chat.Send(message.channel, body: argumentText)
});
```

watchForCommand does the filtering you want, both to make sure it's only getting the command you ask for, and also it provides a convenient `argumentText`.

Reply:
counterpoint: we don't want the base library to contain a command parser

Reply:
my proposal above doesn't do anything to the messages it gets from watchForCommand, so someone could do whatever they want and ignore argumentText. The convenience is just saving 95% of command authors from rewriting the same garbage every time where the (1) check to make sure it's a text message, (2) check to make sure it starts with their command, (3) perform a substring extraction on their command. It would be optional to use, just a pretty function. 

in any case, my proposal doesn't change anything about what you have here, since it's just something new we can add.

Reply:
weirdly enough i can't see your previous comment here, i still think we should do it in another module and make it reusable